src/plugProjectNishimuraU/Armor.cpp

```diff
@@ -675,51 +676,21 @@ void Obj::attackPikmin()
  * Address:	8027E2BC
  * Size:	00008C
  */
-void Obj::getSlotPikiNum()
+int Obj::getSlotPikiNum()
 {
-	/*
-	stwu     r1, -0x20(r1)
-	mflr     r0
-	stw      r0, 0x24(r1)
-	stw      r31, 0x1c(r1)
-	stw      r30, 0x18(r1)
-	li       r30, 0
-	stw      r29, 0x14(r1)
-	stw      r28, 0x10(r1)
-	lwz      r12, 0(r3)
-	lwz      r12, 0x25c(r12)
-	mtctr    r12
-	bctrl
-	lwz      r31, 0(r3)
-	li       r28, 0
-	mr       r29, r3
-	b        lbl_8027E31C
-
-lbl_8027E2FC:
-	mr       r3, r29
-	mr       r4, r28
-	bl       getSlot__10MouthSlotsFi
-	lwz      r0, 0x64(r3)
-	cmplwi   r0, 0
-	beq      lbl_8027E318
-	addi     r30, r30, 1
-
-lbl_8027E318:
-	addi     r28, r28, 1
+	int count = 0;
+	MouthSlots* mouthSlots = getMouthSlots();
+	int max = mouthSlots->getMax();
+	
+	for (int i = 0; i < max; i++){
+		MouthCollPart* part = mouthSlots->getSlot(i);
+
+		if (part->mStuckCreature != nullptr){
```

Comment:
Minor comment, if it doesn't affect the match, could you condense the code into one line?

i.e.
```cpp
if (mouthSlots->getSlot(i)->mStuckCreature)
{
    count++;
}```

Reply:
agreed, looks good otherwise!

Reply:
I think it might affect the match. Not 100% sure so will try asap. The nullptr check could be at least condensed I think ye.

Reply:
All good! Proposed change doesn't affect the match, I commited it. Shall we close the discussion?